### ISSUE TITLE
Map Renders Forecast Zones Incorrectly

### ISSUE THREAD

**Issue by stevekuznetsov** [NWACus/avy#336]
When loading the app for the first time, the map seems to render sometimes in a bottom portion of the screen. After making a selection of the avalanche center, the map stays there.

![File (8)](https://github.com/stevekuznetsov/avalanche-forecast/assets/7328370/dea344df-28a3-4447-9bfc-052813e72013)

Logs from an errant render:

```
{"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0},"windowWidth":0,"windowHeight":0,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":0,"baseOffset":240} output: {"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0}
 INFO  [2023-09-18T14:29:43.335Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":0,"baseOffset":240} output: {"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0}
 INFO  [2023-09-18T14:29:43.336Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":79,"baseOffset":240} output: {"latitude":0,"latitudeDelta":0,"longitude":0,"longitudeDelta":0}
 INFO  [2023-09-18T14:29:43.906Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":79,"baseOffset":240} output: {"latitude":46.80086120781283,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
 INFO  [2023-09-18T14:29:44.319Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":225,"tabBarHeight":79,"baseOffset":240} output: {"latitude":46.80086120781283,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
 INFO  [2023-09-18T14:29:44.349Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":34,"topElementsOffset":477,"cardDrawerMaximumHeight":225,"tabBarHeight":79,"baseOffset":240} output: {"latitude":50.498042834645666,"latitudeDelta":13.291604409448807,"longitude":-121.868,"longitudeDelta":9.021965031556388}
```

Logs from a correct render:

```
 INFO  [2023-09-18T14:44:18.269Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":0,"baseOffset":240} output: {"latitude":47.0964,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
 INFO  [2023-09-18T14:44:18.272Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":0,"tabBarHeight":79,"baseOffset":240} output: {"latitude":46.80086120781283,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
 INFO  [2023-09-18T14:44:18.397Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":0,"topElementsOffset":0,"cardDrawerMaximumHeight":225,"tabBarHeight":79,"baseOffset":240} output: {"latitude":46.80086120781283,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
 INFO  [2023-09-18T14:44:18.428Z] INFO animating map region inputs: {"baseAvalancheCenterMapRegion":{"latitude":47.0964,"latitudeDelta":4.000079999999996,"longitude":-121.868,"longitudeDelta":4.28631},"windowWidth":390,"windowHeight":844,"topElementsHeight":34,"topElementsOffset":55,"cardDrawerMaximumHeight":225,"tabBarHeight":79,"baseOffset":240} output: {"latitude":47.13380997369458,"latitudeDelta":6.314803559644948,"longitude":-121.868,"longitudeDelta":4.28631}
```

**Comment by stevekuznetsov** [comment]
Errant log has `"topElementsOffset":477` whereas normal one has `"topElementsOffset":55,`

**Comment by stevekuznetsov** [comment]
cc @floatplane

**Comment by stevekuznetsov** [comment]
I'm going to reopen this one as there's a clear logic error when we calculate the `topElementsOffset` and I don't know if timing the call to calculate them will be guaranteed to fix that logic error. It still seems like the way we calculate that offset _can_ take into account the size of the AC picker.

**Comment by floatplane** [comment]
are you sure that the 477 is errant? it seems quite possible to me that it was real, with the root cause fixed by #346 (the modal could have rendered a `QueryState` object that took over half the screen, pushing the map downward).

let's see if this repros after we merge #346.

**Comment by stevekuznetsov** [comment]
Errant in the sense that nothing should be rendering there, not that we are calculating the offset incorrectly given the elements.
